src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -506,6 +506,14 @@ auto_flash(std::shared_ptr<xrt_core::device> & workingDevice, const std::string&
     }
     image_path = available_shells.front().second.get<std::string>("file");
   }
+  else if (boost::filesystem::exists(image)) {
+    update_shell(workingDevice->get_device_id(), image, {});
```

Comment:
I figured it out. The issue was due to the flashType is set to old ospi_versal driver instead of xgq driver. I have added the flashType but it seems that it auto_flash will pick the older ospi_versal driver.

Reply:
···

ls /sys/bus/pci/devices/0000\:b3\:00.0/xgq.m.55574528/

/proj/rdi/staff/davidzha/XRT_xocl/build/Debug/runtime_src/core/tools/xbmgmt2/xbmgmt2 program -b --image /root/rpu_01.pdi -d b3:00.0 --flash-type ospi_versal
INFO: ***PDI has 4333808 bytes
XRT build version: 2.13.0
Build hash: 681aa5c5325eec02f83cc84cf665aea7d014cb06
Build date: 2021-11-19 17:37:10
Git branch: versal_gui
PID: 225673
UID: 0
[Mon Nov 22 18:44:56 2021 GMT]
HOST: xsjmaxz50
EXE: /proj/xsjhdstaff2/davidzha/XRT_xocl/build/Debug/runtime_src/core/tools/xbmgmt2/xbmgmt2
[xbmgmt] ERROR: Failed to update base: Invalid argument

···